op-chain-ops/genesis/config.go

```diff
@@ -119,34 +93,50 @@ type DeployConfig struct {
 	L2GenesisBlockGasUsed       hexutil.Uint64 `json:"l2GenesisBlockGasUsed"`
 	L2GenesisBlockParentHash    common.Hash    `json:"l2GenesisBlockParentHash"`
 	L2GenesisBlockBaseFeePerGas *hexutil.Big   `json:"l2GenesisBlockBaseFeePerGas"`
-
-	// L2GenesisRegolithTimeOffset is the number of seconds after genesis block that Regolith hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Regolith.
-	L2GenesisRegolithTimeOffset *hexutil.Uint64 `json:"l2GenesisRegolithTimeOffset,omitempty"`
-	// L2GenesisCanyonTimeOffset is the number of seconds after genesis block that Canyon hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Canyon.
-	L2GenesisCanyonTimeOffset *hexutil.Uint64 `json:"l2GenesisCanyonTimeOffset,omitempty"`
-	// L2GenesisDeltaTimeOffset is the number of seconds after genesis block that Delta hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Delta.
-	L2GenesisDeltaTimeOffset *hexutil.Uint64 `json:"l2GenesisDeltaTimeOffset,omitempty"`
-	// L2GenesisEcotoneTimeOffset is the number of seconds after genesis block that Ecotone hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Ecotone.
-	L2GenesisEcotoneTimeOffset *hexutil.Uint64 `json:"l2GenesisEcotoneTimeOffset,omitempty"`
-	// L2GenesisFjordTimeOffset is the number of seconds after genesis block that Fjord hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Fjord.
-	L2GenesisFjordTimeOffset *hexutil.Uint64 `json:"l2GenesisFjordTimeOffset,omitempty"`
-	// L2GenesisInteropTimeOffset is the number of seconds after genesis block that the Interop hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Interop.
-	L2GenesisInteropTimeOffset *hexutil.Uint64 `json:"l2GenesisInteropTimeOffset,omitempty"`
 	// L2GenesisBlockExtraData is configurable extradata. Will default to []byte("BEDROCK") if left unspecified.
 	L2GenesisBlockExtraData []byte `json:"l2GenesisBlockExtraData"`
+	// Note that there is no L2 genesis timestamp:
+	// This is instead configured based on the timestamp of "l1StartingBlockTag".
+}
+
+var _ ConfigChecker = (*L2GenesisBlockDeployConfig)(nil)
+
+func (d *L2GenesisBlockDeployConfig) Check(log log.Logger) error {
+	if d.L2GenesisBlockGasLimit == 0 {
+		return fmt.Errorf("%w: L2 genesis block gas limit cannot be 0", ErrInvalidDeployConfig)
+	}
+	// When the initial resource config is made to be configurable by the DeployConfig, ensure
+	// that this check is updated to use the values from the DeployConfig instead of the defaults.
+	if uint64(d.L2GenesisBlockGasLimit) < uint64(MaxResourceLimit+SystemTxMaxGas) {
+		return fmt.Errorf("%w: L2 genesis block gas limit is too small", ErrInvalidDeployConfig)
+	}
+	if d.L2GenesisBlockBaseFeePerGas == nil {
+		return fmt.Errorf("%w: L2 genesis block base fee per gas cannot be nil", ErrInvalidDeployConfig)
+	}
+	return nil
+}
+
+type OwnershipDeployConfig struct {
```

Comment:
Probably a good idea to have a godoc on each struct so that its scalable when adding new fields, somebody with less context should be able to look at the godoc and know if its the right place for their config